File: 0374-guess-number-higher-or-lower/0374-guess-number-higher-or-lower.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * int guess(int num);
 */

class Solution {
public:
    int guessNumber(int n) {
        int l = 0, r = n;
        while(true)
        {
            int m = l + (r - l) / 2;
            int a = guess(m);
            if(a == 0)
                return m;
            else if(a == -1)
                r = m - 1;
            else
                l = m + 1;
        }
    }
};

File: 0374-guess-number-higher-or-lower/0374-guess-number-higher-or-lower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if num is higher than the picked number
#          1 if num is lower than the picked number
#          otherwise return 0
# def guess(num: int) -> int:

class Solution:
    def guessNumber(self, n: int) -> int:
        l = 1
        r = n
        while l <= r:
            m = (l + r) // 2
            x = guess(m)
            if x == 1:
                l = m + 1
            elif x == -1:
                r = m - 1
            else:
                return m